src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -70,12 +70,15 @@ export function TopNavMenu(props: TopNavMenuProps): ReactElement | null {
     if (!badges || badges.length === 0) return null;
     return (
       <EuiBadgeGroup className={'kbnTopNavMenu__badgeGroup'}>
-        {badges.map((badge: EuiBadgeProps & { badgeText: string }, i: number) => {
-          const { badgeText, ...badgeProps } = badge;
+        {badges.map((badge: EuiBadgeProps & { badgeText: string, toolTipProps: { content: string; position: string }}, i: number) => {
+          const { badgeText, toolTipProps, ...badgeProps } = badge;
           return (
-            <EuiBadge key={`nav-menu-badge-${i}`} {...badgeProps}>
-              {badgeText}
-            </EuiBadge>
+            <EuiToolTip content={toolTipProps.content} position={toolTipProps.position as ToolTipPositions} 
```

Comment:
```suggestion
            <EuiToolTip {...toolTipProps} 
```

Suggesting that you `spread` the `toolTipProps` so that any tooltip prop passed in from the dashboard level will pass through.

Additionally, you've done it so that the `EuiToolTip` is always wrapping the badge. Does that mean that if `toolTipProps.content` is not given. it will render a blank tooltip? I think so - so it might be best to conditionally wrap the badge if the tooltip is given.

Reply:
@ThomThomson ,
No it will not render blank tooltip, if no tooltip content is set `toolTipProps.content`.